### PR TITLE
Run tests against PostgreSQL 17

### PR DESCRIPTION
**Pull request by IndraGunawan**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| Fixed issues | 

#### Summary

Run CI tests against Postgres 17.

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -242,6 +243,12 @@ jobs:
           - php-version: "8.4"
             postgres-version: "16"
             extension: "pdo_pgsql"
```
Comment (IndraGunawan) [inline .github/workflows/continuous-integration.yml:245]: should i update above `postgres-version` to be `17` ?

**Comment by IndraGunawan** [comment]
should i update `postgres_version` in this list https://github.com/doctrine/dbal/blob/4.2.x/.github/workflows/continuous-integration.yml#L233-L244 to be `17` ?

**Comment by derrabus** [comment]
For 3.9.x please.

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -234,6 +234,7 @@ jobs:
           - "9.4"
           - "15"
           - "16"
+          - "17"
```
Comment (derrabus) [inline .github/workflows/continuous-integration.yml:236]: Please remove 15 from the list.

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -256,6 +257,12 @@ jobs:
           - php-version: "8.4"
             postgres-version: "16"
             extension: "pdo_pgsql"
+          - php-version: "8.4"
+            postgres-version: "17"
+            extension: "pgsql"
+          - php-version: "8.4"
+            postgres-version: "17"
+            extension: "pdo_pgsql"
```
Comment (derrabus) [inline .github/workflows/continuous-integration.yml]: Let's not add more jobs to the CI matrix. It's already too big. All the jobs in the `include` list are supposed to test against the latest Postgres version, so let's instead just replace `16` with `17` in that list.

**Review by derrabus — CHANGES_REQUESTED** [review]

**Review by derrabus — APPROVED** [review]

**Pull request merged**